Add psychemist builder profile page

psychemist opened the pull request:
## Description

Implemented a personal builders page with a retro, game-inspired theme and a simple UI that hides cool easter eggs. Includes interaction with GitHub API to fetch image data.

## Changes

- Added page.tsx that exports my personal builder page component
- Added page.css file that includes styles for specific HTMX elements
- Added resources.ts file that contains seed arrays for implementing UI designs 

## Proof

<img width="705" alt="Screenshot 2025-01-19 at 13 59 48" src="https://github.com/user-attachments/assets/bede51d1-c184-442b-a644-9192d3a8d5d2" />

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

Your ENS/address:
0x8b3C2CB21b6Cb04AE412e41565A804F0bC6D2244

psychemist commented:
ready for review @derrekcoleman

phipsae commented on packages/nextjs/app/builders/0x8b3C2CB21b6Cb04AE412e41565A804F0bC6D2244/page.tsx:
```diff
@@ -0,0 +1,280 @@
+  // Mini-game logic
+  const emojis = ["🌸", "🌺", "🍄", "🌼", "🪻"];
+
+  const handleCellClick = (index: number) => {
+    if (gameBoard[index]) return;
+
+    const newBoard = [...gameBoard];
+    newBoard[index] = emojis[Math.floor(Math.random() * emojis.length)];
+    setGameBoard(newBoard);
+
+    if (selectedCell === null) {
+      setSelectedCell(index);
+    } else {
+      if (newBoard[selectedCell] === newBoard[index]) {
+        setSelectedCell(null);
+      } else {
+        setTimeout(() => {
+          const clearedBoard = [...newBoard];
+          clearedBoard[selectedCell] = "";
+          clearedBoard[index] = "";
+          setGameBoard(clearedBoard);
+          setSelectedCell(null);
+        }, 500);
+      }
+    }
+  };
```
It might makes sense to move the mini game into a separate component

phipsae reviewed (changes requested):
I really appreciate the effort you put into your profile page! 
Well done and so cool, even that you implemented a game, I couldn't figure out how to win xD

Below you'll find some comments, let me know if something is unclear.

psychemist commented:
@phipsae thanks for your comments! i will implement your suggestions and update my PR.

psychemist commented:
## Description

Updated builder profile page with corrections suggested by reviewer @phipsae 

## Changes

- Changed text font color where applicable to enhance readability in dark and light modes.
- Enforced typing and naming convention for builder page component.
- Updated image component from HTML img element to Next Image component for better loading.
- Moved social media links array outside the main page compoenent.
- Separated mini game component into a new file for improved structure.

## Proof

<img width="705" alt="Screenshot 2025-01-19 at 13 59 48" src="https://github.com/user-attachments/assets/bede51d1-c184-442b-a644-9192d3a8d5d2" />

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

Closes #11 

Your ENS/address:
0x8b3C2CB21b6Cb04AE412e41565A804F0bC6D2244

psychemist commented:
Ready for review. Please check and merge. @phipsae

phipsae commented on packages/nextjs/app/builders/0x8b3C2CB21b6Cb04AE412e41565A804F0bC6D2244/page.tsx:
I don't see an implementation of a light mode.

phipsae reviewed (changes requested):
Thanks for all the changes!

Just a tiny comment below and then we can merge.

psychemist commented on packages/nextjs/app/builders/0x8b3C2CB21b6Cb04AE412e41565A804F0bC6D2244/page.tsx:
Alright. Will implement mobile responsiveness and update the. PR. Than ks for the catch!

psychemist commented:
## Description

Implemented responsiveness for inventory and achievement buttons on mobile screens as suggested by reviewer @phipsae  

## Changes

- Adjusted button positions to align with builder profile picture on wide screens and move above atop the page on smaller screens.

## Proof

<img width="386" alt="Screenshot 2025-01-25 at 01 09 45" src="https://github.com/user-attachments/assets/ab21168c-8e0f-4a5e-b767-b72209ed1045" />

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

Closes #11 

Your ENS/address:
0x8b3C2CB21b6Cb04AE412e41565A804F0bC6D2244

psychemist commented:
ready for review @phipsae

psychemist commented:
## Description

Added new UI changes for a more polished, retro gaming look

## Changes

- Added custom font for better readability and style conformity.
- Added background image to enchance user interface.
- Removed animations (moving line and pulsing dots) and accompanying CSS file.

## Proof

<img width="1639" alt="Screenshot 2025-01-25 at 20 44 41" src="https://github.com/user-attachments/assets/13837cbf-f077-4cf8-a310-1789276c4b0b" />


## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

Closes #11 

Your ENS/address:
0x8b3C2CB21b6Cb04AE412e41565A804F0bC6D2244

phipsae commented:
Thanks so much for your effort! Really nice page! It's merged.

psychemist commented:
thank you for all your suggestions! @phipsae
